code/common/analyticslib.q

```diff
@@ -47,7 +47,7 @@ ffill:{[args]
 
 / - General pivot function
 pivot:{[args]
-	checkargs[args;`table`by`piv`var;(" sS")];
+	checkargs[args;`table`by`piv`var`f`g;(" sS")];
```

Comment:
I'm not certain this is the correct fix - I suspect the issue may actually be in the `checkargs` function.

I think the `checkargs` function _should_ be checking that all required keys (2nd arg) are in the dictionary (1st arg). Instead it is currently checking that all keys from the dict are in the list of keys in 2nd arg.

I think making the change here will just enforce that `f` & `g` _must_ be passed (without a type check as there is nothing added to third arg here).

Reply:
Yeah you are right, currently it is checking if all dictionary keys (1st arg) is in the list of required keys, so I think reversing this line `not all key[args] in k` to `not all k in key[args]` should fix this and allow for the optional args to be optional if only the required keys of `table` `by` `piv` `var` are used. This then also allows the 3rd arg to remain the same as it is an `in` check, the alternative I tried would be `~` but `in` makes more sense as the `~` would need to also include the optional args types to work when they are used and that would cause the `~` to fail when they weren't.

Reply:
Yep, sounds good to me 👍